src/System.IO.Pipelines/ReadableBuffer.cs

```diff
@@ -33,7 +31,7 @@ public struct ReadableBuffer : ISequence<ReadOnlyMemory<byte>>
         /// <summary>
         /// Determines if the <see cref="ReadableBuffer"/> is empty.
         /// </summary>
-        public bool IsEmpty => Length == 0;
+        public bool IsEmpty => _first.IsEmpty && Length == 0;
```

Comment:
I have seen it in stress testing my TLS library. Not sure if it was a bug that has gone but I tried to encrypt zero bytes during a foreach over a readable at the start but there was data after that. I had to put an if in that was ugly. Not sure if it is still the case.